### DISCUSSION TITLE
Filepond instead of dropzone?

### DISCUSSION THREAD

**Discussion by sergeitoroptsev** [blakej115/magic-forms#8] [Ideas]
Hey,

In my opinion, it It is more better to use Filepond (https://pqina.nl/filepond/) than Dropzone in UploadForm.

It is already implemented in Magic Forms v2 for Winter CMS (and in my fork as well https://github.com/sergeitoroptsev/oc-forms-plugin).

What You think?

**Comment by blakej115** [comment]
Hi @sergeitoroptsev ,

I will take a look when I have time.

You are also more than welcome to file a pull request. 🙂

Thank you!

**Comment by PubliAlex** [comment]
Hello @blakej115, I worked on a big big recode of the version of magic form originaly coded by skydiver for winter CMS.

It's not just a conversion, I rewrote most of the code to simplify it, add some useful option, use the latest feature of october cms 3.x... That version, like the version 2 of magic form, use filepond to handle file upload.

I created you an access to my private repository to let review the code. If you want, as you took the lead to maintain a version of magic form for october cms users, I can give you the code and let you maintain it.

That version of the module only works for october cms 3.x, it's not compatible with previous versions.

**Comment by PubliAlex** [comment]
@blakej115 Are you ok to let me publish on the october marketplace my version of magic forms based on the V2 (originally for winter cms) ?

**Comment by blakej115** [comment]
@PubliAlex The invitation expired. Do you think you could reinvite me?

It sounds like it could be a good idea to merge it with this version of Magic Forms. I would be more than happy to give you accreditation, as well as give you collaborator access to this repo if it seems fit.

My name is Blake J#2417 on Discord. It is a better way to stay in contact with me.

Thanks!

**Comment by PubliAlex** [comment]
Hello @blakej115 

I just sent you a new invite.

I let you review what I've done as it's a major rewrite, not just a change of upload system and answer me here. If you are interested, I will contact you on discord.

I decided to drop compatibility to october cms before 3.x for this module, so, even if a merge happens, it should be done in a branch exclusive to ocms 3.

Best regards,

**Reply by sergeitoroptsev** [reply]
This is right decision to drop compatibility to octobercms less than v 3. Waiting for new version <3

**Reply by blakej115** [reply]
@PubliAlex Please contact me on Discord. Thank you!

**Comment by blakej115** [comment]
I am currently considering options for major upgrades to the Magic Forms plugin (or creating a spiritual successor).

Thanks!

**Reply by PubliAlex** [reply]
Hello @blakej115,

We were hesitating to release our version of revamped magic form soon now we used it on many projects internally and it seems to be quite stable now.

Are you sure you want to publish a completely revamped version and maintain it ? (if yes, we would prefer to give you the priority because honestly, I'm not sure we will have time to maintain if there are some big feature request). Or, if we publish our, it's ok for you ?

You have an access to our private repository if you want to test the module, if you say it's ok for you, we woule be able to publish it

Best regards,

**Reply by blakej115** [reply]
You are more than welcome to publish your own version of Magic Forms if you wish. It might be good to consider how you will name it to prevent confusion. I am not super interested nor do I personally have the time to invest in maintaining a rewrite of Magic Forms. I am more interested in building my own spiritual successor (when I have time) that would be fundamentally different from Magic Forms.

Thanks!

**Reply by PubliAlex** [reply]
Thanks, we will publish our version of magic form soon

**Comment by PubliAlex** [comment]
Hello,

We released our own implementation of magic form based on the 2.x version that integrate filepond.

We revamped lot of code to simplify some parts, and to add some features when needed.

[https://octobercms.com/plugin/publipresse-forms](https://octobercms.com/plugin/publipresse-forms)

Best regards,
